scripts/main.js

```diff
@@ -0,0 +1,112 @@
+
+const Book = (() => {
+  'user strict';
+
+  class BookObj {
+    constructor(title, author, pages, read) {
+      this.title = title;
+      this.author = author;
+      this.pages = pages;
+      this.read = read;
+    }
+  }
+
+  const setData = () => {
+    const data = localStorage.getItem('data');
+    if (data == null) {
+      localStorage.setItem('data', '[]');
```

Comment:
You can easily parse your data using JSON.stringify([]), instead of doing it manually. 
localStorage.setItem('data', JSON.stringify([]));